### PR TITLE
Added ztfrest metric, modified other metrics, fixed dust extinction

### PR DESCRIPTION
**Pull request by igorandreoni**
This pull request includes all the changes that I have developed in the last few weeks for the kilonova metric, based on your TDE metric and Michael Coughlin's initial modifications. It already includes the fixing of the dust extinction bug. Apologies if this pull request is a bit heavy, future ones should be way smaller. I left two # FIXME at the top because I haven't managed to test those two lines that I suggest to bring back in (they are currently commented out).

### CODE REVIEW

File: mafContrib/kneMetrics.py
Comment (rhiannonlynne) [inline mafContrib/kneMetrics.py:10]: This will break for other users who do not have your dust_values file. 
Another option, which I would prefer, would be to paste your snippet of Dust_values code into this file directly, but call the class something else (My_Dust_values would work), and refer to that. 

On the other hand, Datalab has now updated their copy of MAF and so has Dust_values available as an import. So perhaps this could just entirely be fixed now.
(leaving it like this however will cause `import mafContrib` to fail, just like when a module like george is imported but not available)

File: mafContrib/kneMetrics.py
Comment (rhiannonlynne) [inline mafContrib/kneMetrics.py:73]: Ok .. just for my sanity in the future, 'getLc' would suggest to me fetching a light curve. I think you're meaning it will be returned with the metric value? Would you mind changing this to outputLc or something similar instead?

File: mafContrib/kneMetrics.py
Comment (rhiannonlynne) [inline mafContrib/kneMetrics.py:247]: May I make a suggestion? You don't have to do this but I notice that 'around peak' is similar for most of these detection criteria. Does it make sense to pick out the 'around_peak' indexes first, then pass that subset to each of these selection criteria, instead of the entire dataslice / entire mags / entire time?

File: mafContrib/kneMetrics.py
Comment (rhiannonlynne) [inline mafContrib/kneMetrics.py:317]: would np.random.randint work for you here instead?

File: mafContrib/kneMetrics.py
Comment (rhiannonlynne) [inline mafContrib/kneMetrics.py:31]: I believe this should also be testable on Datalab now :)

**Review by rhiannonlynne — CHANGES_REQUESTED** [review]
One additional thing that would be really useful is a short unit test. 
Is it possible to make up a short runnable example for the metric (metric alone is fine .. another with the slicer would be great) that would give predictable results that we could check? 
Examples of running just a metric -- https://github.com/lsst/sims_maf/blob/master/tests/testCalibrationMetrics.py -- 
If this is something you cannot do, please don't let it hold up the PR. But if you can, it would go a long way toward making sure that any changes in MAF or elsewhere don't break being able to run your metric.

**Comment by rhiannonlynne** [comment]
To be clear, the only thing that really must be fixed before merging is the import, as this will break things for other users. I would really like the renaming of 'getLc' to reduce confusion.  The other items are suggestions.

File: mafContrib/kneMetrics.py
Comment (igorandreoni) [reply mafContrib/kneMetrics.py]: Fixed, all seems to work fine in DataLab.

File: mafContrib/kneMetrics.py
Comment (igorandreoni) [reply mafContrib/kneMetrics.py:68]: 'getLc' is now 'outputLc'

File: mafContrib/kneMetrics.py
Comment (igorandreoni) [reply mafContrib/kneMetrics.py]: This is indeed a smart thing to do, thank you for the suggestion! I modified the code to reduce the number of variables passed.

File: mafContrib/kneMetrics.py
Comment (igorandreoni) [reply mafContrib/kneMetrics.py:303]: Probably yes, but i don't feel 100% confident abut changing this bit (I didn't code it in the first place)

File: mafContrib/kneMetrics.py
Comment (igorandreoni) [reply mafContrib/kneMetrics.py]: Yes, fixed and all good.

**Comment by igorandreoni** [comment]
Thank you for the great feedback, Lynne! I implemented all your suggestions (barred one that I am not sure about). I am working on the unit test, thanks for sending the link to an example. I committed the other changes in the meantime, since it may take a bit longer to create a sensible unit test.

**Pull request merged**
